Src/Newtonsoft.Json/Utilities/MiscellaneousUtils.cs

```diff
@@ -39,15 +39,11 @@ internal static class MiscellaneousUtils
     {
         public static bool ValueEquals(object objA, object objB)
         {
-            if (objA == null && objB == null)
+            if (objA == objB)
             {
                 return true;
             }
-            if (objA != null && objB == null)
-            {
-                return false;
-            }
-            if (objA == null && objB != null)
+            if (objA == null | objB == null)
```

Comment:
Sure. I've gotten into the habit of considering if the RHS is very small whenever I do a `||` or `&&` but that habit may not be widely shared.